### PR TITLE
Qt: Overhaul menubar

### PR DESCRIPTION
**Pull request by spycrab**
Improves the menubar in several ways:
* Add ``...`` to items when appropriate
* Add missing keyboard shortcuts (mainly for ``Open...`` and ``Exit``)

### CODE REVIEW

**Comment by spycrab** [comment]
@ligfx @MayImilae @Starsam80

**Comment by MayImilae** [comment]
Adds icons to entries?

**Comment by spycrab** [comment]
@MayImilae
e.g. a little file icon next to ``Open...``

**Comment by MayImilae** [comment]
That seems a bit unnecessary, imo. The menu bar isn't really conducive to icons, because of how small it is. (It's designed around text)

File: Source/Core/DolphinQt2/MenuBar.cpp [hunk 1]
```diff
@@ -77,22 +77,30 @@ void MenuBar::EmulationStopped()
...
   // Label will be set by a NANDRefresh later
   m_boot_sysmenu = tools_menu->addAction(QStringLiteral(""), [this] { emit BootWiiSystemMenu(); });
   m_boot_sysmenu->setEnabled(false);
 
   connect(&Settings::Instance(), &Settings::NANDRefresh, [this] { UpdateToolsMenu(false); });
 
-  m_perform_online_update_menu = tools_menu->addMenu(tr("Perform Online System Update"));
+  m_perform_online_update_menu =
+      tools_menu->addMenu(QIcon::fromTheme(QStringLiteral("system-software-update")),
+                          tr("Perform Online System Update..."));
```
Comment (ligfx) [inline Source/Core/DolphinQt2/MenuBar.cpp]: This should not have an ellipsis.

**Comment by Starsam80** [comment]
The icons don't seem to appear on windows: 
image

Also, while you are at it, can you also add separators to the `Emulation` menu? This is what WX looks like: 
image

and Qt looks like this: 
image

**Comment by ligfx** [comment]
@Starsam80 you answered my question before I asked it! If the icons only work on Linux, I'd say it's probably not worth the extra code.

**Comment by Starsam80** [comment]
@ligfx No, they do not appear on windows. Quoting the Qt docs:

https://doc.qt.io/qt-5/qicon.html#fromTheme

> Note: By default, only X11 will support themed icons. In order to use themed icons on Mac and Windows, you will have to bundle a compliant theme in one of your themeSearchPaths() and set the appropriate themeName().

I don't think the icons are worth it

**Comment by spycrab** [comment]
@Starsam80 Done.

Now using Dolphin's icons only. Also reduced the amount of icons used due to the fact that our iconset is rather limited.

**Comment by ligfx** [comment]
Ah, sorry, I wasn't clear before: menu icons should not be displayed on macOS.

**Comment by MayImilae** [comment]
Are the icons still in the menu bar on the current version of this?

**Comment by spycrab** [comment]
@MayImilae No, I removed all icons.

File: Source/Core/DolphinQt2/MenuBar.cpp [hunk 2]
```diff
@@ -17,6 +17,7 @@
 #include "DolphinQt2/AboutDialog.h"
 #include "DolphinQt2/GameList/GameFile.h"
 #include "DolphinQt2/MenuBar.h"
+#include "DolphinQt2/Resources.h"
```
Comment (ligfx) [inline Source/Core/DolphinQt2/MenuBar.cpp]: This include isn't needed.

File: Source/Core/DolphinQt2/MenuBar.cpp [hunk 3]
```diff
@@ -77,8 +77,10 @@ void MenuBar::EmulationStopped()
 void MenuBar::AddFileMenu()
 {
   QMenu* file_menu = addMenu(tr("File"));
-  m_open_action = file_menu->addAction(tr("Open"), this, SIGNAL(Open()));
-  m_exit_action = file_menu->addAction(tr("Exit"), this, SIGNAL(Exit()));
+  m_open_action = file_menu->addAction(tr("Open..."), this, SIGNAL(Open()),
+                                       QKeySequence(QStringLiteral("Ctrl+O")));
```
Comment (ligfx) [inline Source/Core/DolphinQt2/MenuBar.cpp]: This should probably use `QKeySequence(QKeySequence::Open)`

File: Source/Core/DolphinQt2/MenuBar.cpp [hunk 4]
```diff
@@ -139,9 +148,9 @@ void MenuBar::AddStateLoadMenu(QMenu* emu_menu)
 void MenuBar::AddStateSaveMenu(QMenu* emu_menu)
 {
   m_state_save_menu = emu_menu->addMenu(tr("Save State"));
-  m_state_save_menu->addAction(tr("Save State to File"), this, &MenuBar::StateSave);
-  m_state_save_menu->addAction(tr("Save State to Selected Slot"), this, &MenuBar::StateSaveSlot);
-  m_state_save_menu->addAction(tr("Save State to Oldest Slot"), this, &MenuBar::StateSaveOldest);
+  m_state_save_menu->addAction(tr("Save State to File..."), this, SIGNAL(StateSave()));
+  m_state_save_menu->addAction(tr("Save State to Selected Slot"), this, SIGNAL(StateSaveSlot()));
+  m_state_save_menu->addAction(tr("Save State to Oldest Slot"), this, SIGNAL(StateSaveOldest()));
```
Comment (ligfx) [inline Source/Core/DolphinQt2/MenuBar.cpp]: These weren't rebased correctly: they should use the compile-time connection syntax.

**Comment by spycrab** [comment]
@dolphin-emu-bot rebuild

**Pull request merged**